Neos.Neos/Tests/Behavior/Features/Bootstrap/ContentRepositorySecurityTrait.php

```diff
@@ -0,0 +1,210 @@
+        $contentGraphProjection = $this->getContentRepositoryService(new class implements ContentRepositoryServiceFactoryInterface {
+            public function build(ContentRepositoryServiceFactoryDependencies $serviceFactoryDependencies): ContentRepositoryServiceInterface
+            {
+                $contentGraphProjection = $serviceFactoryDependencies->projectionsAndCatchUpHooks->contentGraphProjection;
+                return new class ($contentGraphProjection) implements ContentRepositoryServiceInterface {
+                    public function __construct(
+                        public ContentGraphProjectionInterface $contentGraphProjection,
+                    ) {
+                    }
+                };
+            }
+        })->contentGraphProjection;
+        $contentRepositoryAuthProvider = $contentRepositoryAuthProviderFactory->build($this->currentContentRepository->id, $contentGraphProjection->getState());
+
+        TestingAuthProvider::replaceAuthProvider($contentRepositoryAuthProvider);
+        $this->crSecurity_contentRepositorySecurityEnabled = true;
+    }
+
+    private function authenticateAccount(Account $account): void
+    {
+        $this->enableFlowSecurity();
+        $this->crSecurity_testingProvider->setAuthenticationStatus(TokenInterface::AUTHENTICATION_SUCCESSFUL);
+        $this->crSecurity_testingProvider->setAccount($account);
+
+        $securityContext = $this->getObject(SecurityContext::class);
+        $securityContext->clearContext();
+        $securityContext->setRequest($this->crSecurity_mockActionRequest);
+        $this->getObject(AuthenticationProviderManager::class)->authenticate();
+    }
+
+    /**
+     * @Given content repository security is enabled
+     */
+    public function contentRepositorySecurityIsEnabled(): void
+    {
+        $this->enableFlowSecurity();
+        $this->enableContentRepositorySecurity();
+    }
+
+
+    /**
+     * @Given The following additional policies are configured:
+     */
+    public function theFollowingAdditionalPoliciesAreConfigured(PyStringNode $policies): void
+    {
+        $policyService = $this->getObject(PolicyService::class);
+        $policyService->getRoles(); // force initialization
+        $policyConfiguration = ObjectAccess::getProperty($policyService, 'policyConfiguration', true);
+        $mergedPolicyConfiguration = Arrays::arrayMergeRecursiveOverrule($policyConfiguration, Yaml::parse($policies->getRaw()));
+
+        self::$crSecurity_testingPolicyPathAndFilename = $this->getObject(Environment::class)->getPathToTemporaryDirectory() . 'Policy.yaml';
+        file_put_contents(self::$crSecurity_testingPolicyPathAndFilename, Yaml::dump($mergedPolicyConfiguration));
+
+        ObjectAccess::setProperty($policyService, 'initialized', false, true);
+        $this->getObject(ConfigurationManager::class)->flushConfigurationCache();
```

Comment:
Without that I got more failing tests because we have so many runtime caches on all layers

> ... but ill look into it :)

Great, thank you. And let me know if you need support (contentually or mentally *g)